pkg/apis/core/validation/shoot_test.go

```diff
@@ -4892,6 +4902,104 @@ var _ = Describe("Shoot Validation Tests", func() {
 				Expect(ValidateShoot(shoot)).To(BeEmpty())
 			})
 
+			DescribeTable("It should forbid setting certain operation annotations when shoot has a maintenance annotation",
+				func(maintenanceOpAnnotation, operationAnnotation string, errMatcher gomegatypes.GomegaMatcher) {
+					metav1.SetMetaDataAnnotation(&shoot.ObjectMeta, "maintenance.gardener.cloud/operation", maintenanceOpAnnotation)
+					metav1.SetMetaDataAnnotation(&shoot.ObjectMeta, "gardener.cloud/operation", operationAnnotation)
+
+					shoot.Status = core.ShootStatus{
+						LastOperation: &core.LastOperation{
+							Type:  core.LastOperationTypeCreate,
+							State: core.LastOperationStateSucceeded,
+						},
+					}
+
+					if sets.New(v1beta1constants.OperationRotateCredentialsComplete, v1beta1constants.OperationRotateCAComplete, v1beta1constants.OperationRotateServiceAccountKeyComplete, v1beta1constants.OperationRotateETCDEncryptionKeyComplete).Has(maintenanceOpAnnotation) {
+						shoot.Status.Credentials = &core.ShootCredentials{
+							Rotation: &core.ShootCredentialsRotation{
+								CertificateAuthorities: &core.CARotation{
+									Phase: core.RotationPrepared,
+								},
+								ServiceAccountKey: &core.ServiceAccountKeyRotation{
+									Phase: core.RotationPrepared,
+								},
+								ETCDEncryptionKey: &core.ETCDEncryptionKeyRotation{
+									Phase: core.RotationPrepared,
+								},
+							},
+						}
+					}
+
+					Expect(ValidateShoot(shoot)).To(errMatcher)
```

Comment:
```suggestion
					Expect(ValidateShoot(shoot)).To(ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
						"Type":   Equal(field.ErrorTypeForbidden),
						"Field":  Equal("metadata.annotations[gardener.cloud/operation]"),
						"Detail": matcher,
					}))))
```
I am thinking of something like that and passing to the `Entry` function only the `ContainSubstring` matcher